File: app/src/main/java/com/chooloo/www/callmanager/ui/fragment/settings/SettingsPresenter.java
package com.chooloo.www.callmanager.ui.fragment.settings;

import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.OnLifecycleEvent;
import androidx.preference.Preference;

import com.chooloo.www.callmanager.ui.base.BaseContract;
import com.chooloo.www.callmanager.ui.base.BasePresenter;
import com.chooloo.www.callmanager.util.PermissionUtils;

public class SettingsPresenter<V extends SettingsContract.View> extends BasePresenter<V> implements SettingsContract.Presenter<V> {

    @Override
    public void onRequestPermissionResult(int requestCode, int[] grantResults) {
        if (requestCode == PermissionUtils.PERMISSION_RC && PermissionUtils.checkPermissionsGranted(grantResults)) {
            mView.setupSimSelection();
        }
    }

    @Override
    public void refresh() {
        mView.goToMainActivity();
    }

    @Override
    public boolean onListPreferenceChange(Preference preference, Object newValue) {
        mView.setListPreferenceSummary(preference, newValue);
        return true;
    }

    @Override
    public boolean onSwitchPreferenceChange(Preference preference, Object newValue) {
        mView.setSwitchPreferenceSummary(preference, newValue);
        return true;
    }

    @Override
    public boolean onCheckBoxPreferenceChange(Preference preference, Object newValue) {
        mView.setCheckBoxPreferenceSummary(preference, newValue);
        return true;
    }
}

File: app/src/main/java/com/chooloo/www/koler/ui/settingsfragment/SettingsPresenter.kt
package com.chooloo.www.koler.ui.settingsfragment

import com.chooloo.www.koler.ui.base.BasePresenter

class SettingsPresenter<V : SettingsFragmentContract.View> : BasePresenter<V>(),
    SettingsFragmentContract.Presenter<V> {
    override fun refresh() {
        mvpView?.goToMainActivity()
    }

    override fun onSimSelectionChanged(newValue: Any?) {
        TODO("Not yet implemented")
    }

    override fun onAppThemeSelectionChanged(color: Int) {
        TODO("Not yet implemented")
    }
}